### PR TITLE
GH-218: Adapt the occupation-inheritance flow cap to tree density

### LINKED ISSUE #218: Compute the occupation-inheritance flow cap adaptively instead of a hard-coded limit
## Context

The Overview-tab parent → child occupation Sankey (`OccupationInheritanceRepository::occupationInheritance()`) caps the displayed flows at a **hard-coded 12** passed from the template.

## Problem

The flow-weight distribution is highly skewed: a long tail of one-off (weight-1) parent-trade → child-trade pairs dominates the flow *count*, while a handful of recurring flows carry the signal. A fixed top-N is wrong in both directions:

- **Dense tree** — meaningful recurring flows just past the cut are truncated.
- **Sparse tree** — the diagram is padded with single-occurrence noise.

Naive "dynamic" schemes fail on this shape:

- *fraction-of-max* is dominated by a single high-weight outlier (one strong dynasty pulls the threshold up and erases the mid-tier);
- *cumulative-coverage / Pareto* pulls in almost the whole tail, because the many one-offs are collectively the majority of total weight.

## Proposal

Separate the **rendering ceiling** from the **data floor**:

- **Legibility ceiling** (`MAX_FLOWS`) stays a fixed constant — a Sankey beyond ~15 flows is an unreadable hairball regardless of tree size. This is a property of the medium, not the data.
- **Data floor** via a minimum flow weight (`MIN_FLOW_WEIGHT`): drop pure one-off pairs — an inheritance *pattern* requires recurrence in at least two families. This self-adapts: a sparse tree shows fewer flows, a dense tree fills up to the ceiling.
- A tree with **no recurring** occupational inheritance yields an **honestly empty** diagram rather than a wall of one-off noise.

Both thresholds become named constants in the repository; the magic number is removed from the template. The raw aggregator stays parameterisable so the data-layer tests can still exercise unfiltered weighting.

### PR DESCRIPTION
**Pull request by magicsunday**
## Summary

The Overview-tab parent → child occupation-inheritance Sankey capped its displayed flows at a **hard-coded 12** passed from the template. The flow-weight distribution is highly skewed — a long tail of one-off (weight-1) pairs dominates the flow count — so a fixed top-N truncates a dense tree while padding a sparse one with single-occurrence noise.

This replaces the magic number with an **adaptive display policy**, split into a rendering ceiling and a data floor, both named constants in `OccupationInheritanceRepository`:

- **`MAX_FLOWS` (15)** — a fixed legibility ceiling. A Sankey beyond ~15 flows is an unreadable hairball regardless of tree size; this is a property of the medium, not the data.
- **`MIN_FLOW_WEIGHT` (2)** — drops one-off flows. A flow's weight counts the *children* exhibiting it, so this keeps only patterns that recur across at least two children. This is the data-driven floor that lets the shown-flow count adapt to a tree's density; a tree with no recurring occupational inheritance now yields an **honestly empty** diagram.

Naive alternatives were rejected because they fail on this distribution shape: fraction-of-max is dominated by a single high-weight outlier, and cumulative-coverage/Pareto pulls in nearly the whole tail (the many one-offs are collectively the majority of total weight).

## Design note

`occupationInheritance()` gains `$maxFlows` and `$minWeight` parameters that default to the two constants, so:
- the consumed path (`Statistic::getOccupationInheritance()` → template) applies the policy with **no magic number in the template**;
- the data-layer integration tests pass `minWeight = 1` to exercise the **unfiltered** weighting unchanged (a guarded `if ($minWeight > 1)` keeps that path byte-identical).

## Tests

- Two new integration tests: one asserts one-off flows are dropped (only the recurring flow survives), one asserts the empty-diagram behavior when no flow recurs.
- The 12 existing aggregation tests are unchanged in behavior (explicit `minWeight = 1`).
- Full suite green (phpstan / cgl / rector / 813 unit + MySQL integration). Verified live against a real tree: 15 legible flows, one-off noise dropped.

Closes #218

### CODE REVIEW

File: src/Repository/OccupationInheritanceRepository.php [hunk 1]
```diff
@@ -103,10 +130,13 @@ public function __construct(
...
-     * @param int $topLinks Maximum number of distinct flows to retain
+     * @param int $maxFlows  Legibility ceiling on the number of flows returned
+     * @param int $minWeight Minimum weight a flow must reach to be shown; 1 returns every flow unfiltered
      */
-    public function occupationInheritance(int $topLinks): SankeyFlowsPayload
-    {
+    public function occupationInheritance(
+        int $maxFlows = self::MAX_FLOWS,
+        int $minWeight = self::MIN_FLOW_WEIGHT,
+    ): SankeyFlowsPayload {
```
[the remark this answers is not in the document]
Comment (magicsunday) [reply src/Repository/OccupationInheritanceRepository.php:139]: Verified against the call sites — these two parameters carry no untrusted or negative input in practice. The sole production caller, `Statistic::getOccupationInheritance()`, passes no arguments, so the shipped path always uses the `MAX_FLOWS` / `MIN_FLOW_WEIGHT` constants; the only other callers are the integration tests passing controlled values (`10, 1` / `1, 1`).

The edge values are also non-harmful:
- `minWeight <= 1` takes the `if ($minWeight > 1)` false branch and returns every flow unfiltered — exactly the documented `minWeight = 1 returns every flow` contract — so `max(1, ...)` would be a behavioural no-op.
- a zero/negative `maxFlows` only produces a well-defined `array_slice` result (empty for `0`, a partial slice for negatives) with no crash, leak, or malformed payload.

Since no reachable caller can pass a harmful value, the clamp would be speculative defensive code for a non-manifesting case, so I am leaving the signature unchanged. Thanks for the check.
